domain/build.gradle

```diff
@@ -26,6 +26,9 @@ android {
 
   kotlinOptions {
     jvmTarget = JavaVersion.VERSION_1_8
+    freeCompilerArgs += [
+        "-Xuse-experimental=kotlinx.coroutines.ExperimentalCoroutinesApi"
```

Comment:
Why are we using experimental API ? @BenHenning @rt4914 can we use it ?

Reply:
@aggarwalpulkit596 Not sure about this. Will need to defer to @BenHenning on this.

Reply:
I think it's necessary here since otherwise it's quite difficult to use Deferred. We also technically use the experimental API throughout the codebase (mostly for tests), and it seems pretty stable since it's been around for a few years now. This seems fine to me.

However, @prayutsu we should also add this argument to the Bazel builds for tests & binaries, too.

Reply:
@prayutsu reminder.

Reply:
@MaskedCarrot Kindly follow up with @prayutsu and make sure it get resolved.

Reply:
Sorry, I'm not sure why I kept at-mentioning Abhay. @MaskedCarrot I meant to at-mention you. This needs to be resolved.